### ISSUE TITLE
Can this be open sourced?

### ISSUE THREAD

**Issue by amithegde** [jozefizso/apimonitor#1]
Hi @rohitab, can this tool be made open source? Or a commercial tool?

**Comment by rohitab** [comment]
@amithegde, it's been a while since I've had time to work on API Monitor. But, I do have plans to resume development and add additional features. As for making it open source, I haven't made a decision yet.

**Comment by amithegde** [comment]
This tool saves so much time while debugging large applications and to get a birds eye view. I have few feature suggestions if you consider resuming development!

**Comment by rohitab** [comment]
@amithegde of course. I'm very open to suggestions and would be happy to add new features in once I start working on it again.

Thanks for the feedback too. Glad to know you're finding it useful.

**Comment by amithegde** [comment]
Here are few that help:

- Can we make the process tree filterable? As of now I sort by name and try to attach, but if there are many processes getting crated, it keeps jumping around
- Can we have a configuration to specify auto attach for a process? That way, as soon as ApiMonitor becomes aware of the process, it automatially attaches to it. This is because we have to let the application run in its own context, activated by another process and so on.
- For the APIs captured, can we have a filter at the top which allows filtering results better? Currently I try to go by thread or module, but if I an filter quickly by entering filte conditions, that would make it much better
- Can we allow exporing captured results to Excel which allows slicing/dicing data better? For example, I was debugging through pdh.dll to figure out some perf count issue, and there were thousands of calls and if there is better way to filter, that makes it easy.
- Can we have a console version? That autmatically captures traces to a file which we can explore later on?

**Comment by rohitab** [comment]
Great ideas. I have most of these on my todo list, so they will be implemented for sure. I had been holding of on developing some features, like exporting to a variety of file formats, since I wanted to implement them via plugins, either mine or 3rd party. I also plan to add scripting, so API Monitor can be fully automated.

**Comment by amithegde** [comment]
thank you @rohitab plugins and ability to script are great.

Additionally, I think currenlty there is no way to provide symbols band because of which the stack doesn't resolve properly. Even when I have private symbols, it is kind of hard to view full call stack.

A good example to look at would be [perfview ](https://github.com/microsoft/perfview) which allows folding callstack and expanding call stack and also allows loading symbols:
![image](https://github.com/user-attachments/assets/8290490f-9956-4363-a746-d6a0a03f5063)

May be if calls collected can be formatted to make it load on JS version of Perfview which would save the effort required to write the UI - https://github.com/microsoft/perfview/issues/903

If we have these capabilities, the tool can be extremely useful to debug various scenarios which are tedios to debug through.

**Comment by rohitab** [comment]
@amithegde, loading symbols is definitely required and is already on the roadmap. I'm not familiar with PerfViewJS, but I've bookmarked this page, so when I start implementing these features, I'll get back to you and we can discuss it at that time.

**Comment by SmallMouseClimbs** [comment]
I'm very glad to see that you continue to update this tool. However, some APIs are not fully supported. For example, when I'm monitoring OpenGL, quite a number of functions need to be written by myself. It would be great if they could be updated. Thank you again for developing this tool software.

**Comment by ohault** [comment]
Hi @rohitab, I could argue that going Open Source would enable a smart way to explore how to use API monitor via Virtual Machine Introspection (VMI) ?

**Comment by hardliner66** [comment]
> Hi [@rohitab](https://github.com/rohitab), I could argue that going Open Source would enable a smart way to explore how to use API monitor via Virtual Machine Introspection (VMI) ?

It would also allow others to contribute to the project. Especially knowledge and function definitions.
